rust/xaynet-analytics/src/database/instance.rs

```diff
@@ -0,0 +1,116 @@
+use isar_core::{
+    collection::IsarCollection,
+    error::IsarError,
+    instance::IsarInstance,
+    object::{object_builder::ObjectBuilder, object_id::ObjectId},
+    schema::{collection_schema::CollectionSchema, Schema},
+};
+use std::vec::IntoIter;
+
+use crate::database::{
+    analytics_event::data_model::AnalyticsEvent,
+    common::{FieldProperty, IsarAdapter},
+};
+
+pub struct DbInstance {
```

Comment:
Since the rest of the code should use the `Repo` trait to interact with the database we can call This `IsarDB` or something similar. (Also the name of the file can be changed to `isar`).

Reply:
sure will do, makes sense